### PR TITLE
DSD-1181: update icon colors

### PR DESCRIPTION
**Pull request by oliviawongnyc**
Fixes JIRA ticket [DSD-1181.](https://jira.nypl.org/browse/DSD-1181)

## This PR does the following:

- Adds requested color options to the Icon component.
- Adds IconColors and IconNames files to cut down on repetitive lists.

## How has this been tested?

- In Storybook.

## Accessibility concerns or updates

- Are there dark versions of the section colors?

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
 
 ## Prerelease
 
+### Adds
+
+- Adds additional color options to `Icon` component.
```
Comment (EdwinGuzman) [inline CHANGELOG.md:13]: Add that the type/array structure was updated as well to make it easier to update.

File: src/components/Icons/Icon.stories.mdx [hunk 2]
```diff
@@ -32,27 +34,7 @@ import { getCategory } from "../../utils/componentCategories";
...
-        "section.locations.secondary",
-        "section.research.primary",
-        "section.research.secondary",
-        "section.research-library.lpa",
-        "section.research-library.schomburg",
-        "section.research-library.schwartzman",
-        "section.whats-on.primary",
-        "section.whats-on.secondary",
-      ],
+      options: getIconColors(),
```
Comment (EdwinGuzman) [inline src/components/Icons/Icon.stories.mdx:37]: Do the same for the `name` prop so we can update it with the Controls.

File: src/components/Icons/IconNames.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  "socialInstagram",
+  "socialTumblr",
+  "socialTwitter",
+  "socialYoutube",
+  "speakerNotes",
+  "utilityAccountFilled",
+  "utilityAccountUnfilled",
+  "utilityHamburger",
+  "utilitySearch",
+];
```
Comment (EdwinGuzman) [inline src/components/Icons/IconNames.tsx]: Why does this not need the `as const`?

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
Looks good! Just one minor request and a few questions.

File: src/components/Icons/IconNames.tsx (hunk 3 above)
Comment (oliviawongnyc) [reply src/components/Icons/IconNames.tsx]: Um not sure. I left it out accidentally. Just added.

**Review by EdwinGuzman — APPROVED** [review]

**Review by bigfishdesign13 — APPROVED** [review]
This looks good, but why is the description for the `color` prop empty on the Storybook Docs view?

**Comment by oliviawongnyc** [comment]
> This looks good, but why is the description for the `color` prop empty on the Storybook Docs view?

Very strange. This is true on prod as well. I looked at the code and cannot understand why this is the only description not pulling through.

**Pull request merged**